core/vm/evm.go

```diff
@@ -202,12 +202,13 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 	if !evm.StateDB.Exist(addr) {
 		if !isPrecompile && evm.chainRules.IsEIP4762 {
 			// add proof of absence to witness
-			wgas := evm.Accesses.TouchFullAccount(addr.Bytes(), false)
-			if gas < wgas {
+			gc := gasConsumer{availableGas: gas}
+			ok := evm.Accesses.TouchFullAccount(addr.Bytes(), false, gc.consumeGas)
+			if !ok {
 				evm.StateDB.RevertToSnapshot(snapshot)
-				return nil, 0, ErrOutOfGas
+				return nil, gc.availableGas, ErrOutOfGas
```

Comment:
@gballet, I think this was a bug before, and this PR allows us to do the correct thing.
LMK your thoughts.